src/normalize/structure.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 OpenStax Poland
+// Licensed under the MIT license. See LICENSE file in the project root for
+// full license text.
+
+import { Editor, NodeEntry, Transforms } from 'slate'
+
+import { Glossary, Section, Title } from '../interfaces'
+
+/**
+ * Normalize document structure
+ *
+ * Return true if entry was normalized and false otherwise.
+ */
+export default function normalizeStructure(editor: Editor, entry: NodeEntry): boolean {
+    const [node, path] = entry
+
+    // The document.
+    if (path.length === 0) {
+        // Document must not be empty and must not start with a glossary.
```

Comment:
I think we should also check if there isn't anything after `glossary`

Reply:
Normalization for glossary ensures that it's always the last element, so additional check here is unnecessary.